src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/time/utils.tsx

```diff
@@ -0,0 +1,113 @@
+import moment from 'moment';
+
+import {use24Hours} from 'app/utils/dates';
+import {defined} from 'app/utils';
+import {t} from 'app/locale';
+
+enum TimeAbbreviation {
+  SECOND = 'second',
+  MINUTE = 'minute',
+  HOUR = 'hour',
+  DAY = 'day',
+  YEAR = 'year',
+}
+
+const timeFormat = 'HH:mm:ss';
+const timeDateFormat = `ll ${timeFormat}`;
+
+const getTimeShortString = (type: TimeAbbreviation, time: number) => {
+  switch (type) {
+    case TimeAbbreviation.SECOND:
+      return t('%s sec', time);
+    case TimeAbbreviation.MINUTE:
+      return t('%s min', time);
+    case TimeAbbreviation.HOUR:
+      return t('%s hr', time);
+    case TimeAbbreviation.DAY:
+      return t('%s d', time);
+    case TimeAbbreviation.YEAR:
+      return t('%s d', time);
+    default:
+      return '';
+  }
+};
+
+const getShortRelativeTime = (milliseconds: number) => {
+  const seconds = Math.round(milliseconds / 1000);
+  const minutes = Math.round(seconds / 60);
+  const hours = Math.round(minutes / 60);
+  const days = Math.round(hours / 24);
+  const years = Math.round(days / 365);
+
+  const args = ((seconds < 45 && [TimeAbbreviation.SECOND, seconds]) ||
+    (minutes < 45 && [TimeAbbreviation.MINUTE, minutes]) ||
+    (hours < 22 && [TimeAbbreviation.HOUR, hours]) ||
+    (days <= 300 && [TimeAbbreviation.DAY, days]) || [TimeAbbreviation.YEAR, years]) as [
+    TimeAbbreviation,
+    number
+  ];
+
+  return getTimeShortString(args[0], args[1]);
+};
+
```

Comment:
Can we re-use something in `app/utils/formatters`?

Reply:
I will move the utils methods in a follow-up PR

Reply:
good idea 💡